src/main/java/com/epam/ta/reportportal/core/project/settings/impl/DeleteProjectSettingsHandler.java

```diff
@@ -0,0 +1,121 @@
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+@Service
+@Transactional
+public class DeleteProjectSettingsHandler implements IDeleteProjectSettingsHandler {
+
+	private ProjectRepository projectRepository;
+
+	private StatisticsFieldRepository statisticsFieldRepository;
+
+	private WidgetRepository widgetRepository;
+
+	private MessageBus messageBus;
+
+	private IssueTypeRepository issueTypeRepository;
+
+	private IssueEntityRepository issueEntityRepository;
+
+	@Autowired
+	public DeleteProjectSettingsHandler(ProjectRepository projectRepository, StatisticsFieldRepository statisticsFieldRepository,
+			WidgetRepository widgetRepository, MessageBus messageBus, IssueTypeRepository issueTypeRepository,
+			IssueEntityRepository issueEntityRepository) {
+		this.projectRepository = projectRepository;
+		this.statisticsFieldRepository = statisticsFieldRepository;
+		this.widgetRepository = widgetRepository;
+		this.messageBus = messageBus;
+		this.issueTypeRepository = issueTypeRepository;
+		this.issueEntityRepository = issueEntityRepository;
+	}
+
+	@Override
+	public OperationCompletionRS deleteProjectIssueSubType(ReportPortalUser.ProjectDetails projectDetails, ReportPortalUser user, Long id) {
+		Project project = projectRepository.findById(projectDetails.getProjectId())
+				.orElseThrow(() -> new ReportPortalException(PROJECT_NOT_FOUND, projectDetails.getProjectId()));
+
+		IssueType type = project.getIssueTypes()
+				.stream().filter(issueType -> issueType.getId().equals(id))
+				.findFirst()
+				.orElseThrow(() -> new ReportPortalException(ISSUE_TYPE_NOT_FOUND, id));
+
+		if (Sets.newHashSet(
+				AUTOMATION_BUG.getLocator(),
+				PRODUCT_BUG.getLocator(),
+				SYSTEM_ISSUE.getLocator(),
+				NO_DEFECT.getLocator(),
+				TO_INVESTIGATE.getLocator()
+		).contains(type.getLocator())) {
+			fail().withError(FORBIDDEN_OPERATION, "You cannot remove predefined global issue types.");
+		}
+
+		String issueField = "statistics$defects$" + TestItemIssueGroup.fromValue(type.getIssueGroup().getTestItemIssueGroup().getValue())
+				.orElseThrow(() -> new ReportPortalException(ISSUE_TYPE_NOT_FOUND, type.getIssueGroup()))
+				.getValue()
+				.toLowerCase() + "$" + type.getLocator();
+		statisticsFieldRepository.deleteByName(issueField);
+
+		IssueType defaultGroupIssueType = issueTypeRepository.findByLocator(type.getIssueGroup().getTestItemIssueGroup().getLocator());
+		List<IssueEntity> allByIssueTypeId = issueEntityRepository.findAllByIssueTypeId(id);
+		allByIssueTypeId.forEach(issueEntity -> issueEntity.setIssueType(defaultGroupIssueType));
+
+		project.getIssueTypes().remove(type);
+		projectRepository.save(project);
+
+		widgetRepository.findAllByProjectId(projectDetails.getProjectId()).stream().filter(widget -> {
+			String widgetType = widget.getWidgetType();
+			return widgetType.equals(LAUNCHES_TABLE.getType()) || widgetType.equals(STATISTIC_TREND.getType());
```

Comment:
possible npe
write in one line: LAUNCHES_TABLE.getType().equals(widget.getWidgetTYpe)